### PR TITLE
[Merged by Bors] - feat: port Data.Sigma.Lex

### PR DESCRIPTION
**Pull request by riccardobrasca**
mathlib3 sha: fd47bdf09e90f553519c712378e651975fe8c829

### CODE REVIEW

File: Mathlib/Data/Sigma/Lex.lean [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+      · exact Or.inr (Or.inr $ Lex.right _ _ hba)
+
+
+    · exact Or.inr (Or.inr $ Lex.left _ _ hji)
+      ⟩
+
+end Sigma
+
+/-! ### `psigma` -/
```
Comment (mcdoll) [inline Mathlib/Data/Sigma/Lex.lean:174]:
```suggestion
/-! ### `PSigma` -/
```

File: Mathlib/Data/Sigma/Lex.lean [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+## See also
+
+--Porting note: fix these once the files are ported.
+Related files are:
+* `data.finset.colex`: Colexicographic order on finite sets.
+* `data.list.lex`: Lexicographic order on lists.
+* `data.sigma.order`: Lexicographic order on `Σ i, α i` per say.
+* `data.psigma.order`: Lexicographic order on `Σ' i, α i`.
+* `data.prod.lex`: Lexicographic order on `α × β`. Can be thought of as the special case of
+  `sigma.lex` where all summands are the same
```
Comment (mcdoll) [inline Mathlib/Data/Sigma/Lex.lean:28]:
```suggestion
Related files are:
* `Data.Finset.Colex`: Colexicographic order on finite sets.
* `Data.List.Lex`: Lexicographic order on lists.
* `Data.Sigma.Order`: Lexicographic order on `Σ i, α i` per say.
* `Data.PSigma.Order`: Lexicographic order on `Σ' i, α i`.
* `Data.Prod.Lex`: Lexicographic order on `α × β`. Can be thought of as the special case of
  `Sigma.Lex` where all summands are the same
```
I am not sure whether it is `Colex` or `CoLex`, but all other names are easy and I think we should change them now. Also that file seems to non-existing in mathlib3 (maybe moved to `combinatorics.colex`?)

**Review by Ruben-VandeVelde — APPROVED** [review]
lgtm

File: Mathlib/Data/Sigma/Lex.lean [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+#align sigma.lex Sigma.Lex
+
+theorem lex_iff : Lex r s a b ↔ r a.1 b.1 ∨ ∃ h : a.1 = b.1, s b.1 (h.rec a.2) b.2 := by
+  constructor
+  · rintro (⟨a, b, hij⟩ | ⟨a, b, hab⟩)
+    · exact Or.inl hij
+
+    · exact Or.inr ⟨rfl, hab⟩
+
+
```
Comment (j-loreaux) [inline Mathlib/Data/Sigma/Lex.lean:51]: I think the extra spacing here and elsewhere is not necessary. Can you remove it?

File: Mathlib/Data/Sigma/Lex.lean [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+theorem Lex.mono (hr : ∀ a b, r₁ a b → r₂ a b) (hs : ∀ i a b, s₁ i a b → s₂ i a b) {a b : Σ i, α i}
+    (h : Lex r₁ s₁ a b) : Lex r₂ s₂ a b := by
+  obtain ⟨a, b, hij⟩ | ⟨a, b, hab⟩ := h
+  · exact Lex.left _ _ (hr _ _ hij)
+
+  · exact Lex.right _ _ (hs _ _ _ hab)
+
+#align sigma.lex.mono Sigma.Lex.mono
+
+theorem Lex.monoLeft (hr : ∀ a b, r₁ a b → r₂ a b) {a b : Σ i, α i} (h : Lex r₁ s a b) :
```
Comment (j-loreaux) [inline Mathlib/Data/Sigma/Lex.lean:77]: This should be called `Sigma.Lex.mono_left` and similarly for `monoRight`.

**Review by j-loreaux — COMMENTED** [review]
Minor comments, but overall looks good.

**Comment by semorrison** [comment]
bors merge

**Comment by riccardobrasca** [comment]
bors merge

**Pull request closed without merging**